pkg/util/cloudproviders/cloudfoundry/cccache.go

```diff
@@ -171,6 +180,44 @@ func (ccc *CCCache) UpdatedOnce() <-chan struct{} {
 	return ccc.updatedOnce
 }
 
+func (ccc *CCCache) waitForResource(guid string) {
+	ccc.RLock()
+	ch, ok := ccc.activeResources[guid]
+	ccc.RUnlock()
+	if ok {
+		// wait for the resource to be released
+		<-ch
+	}
+}
+
+func (ccc *CCCache) setResourceActive(guid string) error {
```

Comment:
I think you can combine wait and set active, possibly removing the error condition, to simplify things.

Reply:
The reason I didn't combine them is that I need to check the cache before setting the resource active. I actually forgot to add that check because the tests passed.